src/linker/Linker.Dataflow/ArrayElementReferenceValue.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System.Linq;
+using ILLink.Shared.DataFlow;
+using MultiValue = ILLink.Shared.DataFlow.ValueSet<ILLink.Shared.DataFlow.SingleValue>;
+
+namespace ILLink.Shared.TrimAnalysis
+{
+	public partial record ArrayElementReferenceValue (MultiValue ReferencedValue) : ReferenceValue
```

Comment:
Can we not add this since we don't really support it?
I would simply store an unknown value onto the stack when `ldelema` is used. It should be VERY uncommon and I'm fine warning there always.